src/core/hardage.rs

```diff
@@ -0,0 +1,31 @@
+use crate::error;
+
+pub(crate) fn hardage(c: char) -> error::Result<i32> {
+    match c {
+        'A'..='Z' | 'a'..='z' => Ok(1),
+        '0' | '4' | '5' | '6' => Ok(2),
+        '1' | '2' | '3' | '7' | '8' | '9' => Ok(4),
+        '-' => Err(Box::new(error::Error::UnexpectedCode(
+            "count code start".to_owned(),
+        ))),
+        '_' => Err(Box::new(error::Error::UnexpectedCode(
+            "op code start".to_owned(),
+        ))),
+        _ => Err(Box::new(error::Error::UnknownHardage(c.to_string()))),
```

Comment:
I typically use `to_owned()` for strings because I read there was potentially less overhead. Not sure if it's the right thing to do.

Reply:
Just seeing this now.

Reply:
Resolved with #31